File: brain_tumor_front/src/types/role.ts
// Role 타입 정의(역할 정의)
// TODO : DB에서 Role 어떻게 받아오는지 확인 후 코드 수정
export type Role =
    |'SYSTEMMANAGER' // 시스템 관리자
    |'ADMIN' // 관리자
    |'DOCTOR' // 의사
    |'NURSE' // 간호사
    |'RIS' // 영상과
    |'LIS' // 검사과
    |'PATIENT' // 환자
;

File: brain_tumor_back/apps/accounts/models/role.py
from django.db import models

# Role 모델
class Role(models.Model):
    code = models.CharField(max_length=50, unique = True) # Doctor, Nurse, Patient, LIS, RIS, Admin
    name = models.CharField(max_length=50)
    description = models.TextField(blank= True)
    created_at = models.DateTimeField(auto_now_add = True)
    
    def __str__(self) :
        return self.name

